### PR TITLE
Add support for managed identities

### PR DESCRIPTION
**Pull request by nikola-naydenov-hmcts**
Adding MI support for access to service bus topics. Needs testing.
Please suggest any improvements.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -47,3 +47,32 @@ variable "requires_session" {
   description = "A value that indicates whether the queue supports the concept of sessions"
   default     = false
 }
+
+variable "create_managed_identity" {
+  default = false
+}
```
Comment (timja) [inline variables.tf:53]: I don't think this is needed here, everyone generally has created as part of key vault module, probably don't want to add that in multiple places?

Comment (nikola-naydenov-hmcts) [reply variables.tf]: Yeah I wasn't sure. I have removed the create bits.

File: variables.tf [hunk 2]
```diff
@@ -47,3 +47,13 @@ variable "requires_session" {
   description = "A value that indicates whether the queue supports the concept of sessions"
   default     = false
 }
+
+variable "managed_identity_object_ids" {
```
Comment (timja) [inline variables.tf:51]: do we need 2 ways of doing this? just one should be fine? i.e. ids vs id

Comment (nikola-naydenov-hmcts) [reply variables.tf:51]: This I borrowed from the key vault code. I kinda went in blind here so I wanted to try follow an example from other platops terraform modules.

I can change it to just ids if you feel strongly about it.

Comment (timja) [reply variables.tf:51]: i would just do ids, this is clean slate whereas that evolved over many years

**Review by timja — APPROVED** [review]

**Pull request merged**